x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/action/filter/ShardBulkInferenceActionFilter.java

```diff
@@ -416,36 +416,41 @@ private Map<String, List<FieldInferenceRequest>> createFieldInferenceRequests(Bu
                 for (var entry : fieldInferenceMap.values()) {
                     String field = entry.getName();
                     String inferenceId = entry.getInferenceId();
-                    Object inferenceResult = inferenceMap.remove(field);
-                    var value = XContentMapValues.extractValue(field, docMap);
-                    if (value == null) {
-                        if (inferenceResult != null) {
+                    for (var sourceField : entry.getSourceFields()) {
+                        Object inferenceResult = inferenceMap.remove(field);
```

Comment:
If there are multiple source fields, doesn't this logic remove `field` from `inferenceMap` multiple times? Is this an issue?